core/linux.c

```diff
@@ -83,18 +83,17 @@ const char *system_default_directory(void)
 
 const char *system_default_filename(void)
 {
-	static char *filename = NULL;
-	if (!filename) {
```

Comment:
one cannot comment on the commit message, which is a little hard to parse for me.
But obviously the filename can be NULL since it's a pointer.